pennylane/transforms/transpile.py

```diff
@@ -126,7 +126,7 @@ def circuit():
     """
     if device:
         device_wires = device.wires
-        is_default_mixed = getattr(device, "short_name", "") == "default.mixed"
+        is_default_mixed = getattr(device, "name", "") == "default.mixed"
```

Comment:
Though also, why does transpile care about whether or not it's `default.mixed`? This seems a bit smelly and should probalby be fixed.

Reply:
Yeah similar things spread over the codebase. This place is at least explicitly pointing out `default.mixed`. Many things seem to be implicity specialized for the legacy `default.mixed`. For example, the legacy custom_decomps branch in the device constructor.

Reply:
Searching over this file, this boolean seems to be used for transforming `StateMP` into `DensityMatrixMP` on `default.mixed`. Feels necessary if we want to keep supporting the shortcut "state for density matrix when talking about mixed device".

Reply:
@albi3ro any thoughts on this? Right now my tendency is to keep it since it supports our `state()` being transformed to `DensityMatrxMP`

Reply:
I'm not a huge fan of this, but I also don't feel like dealing with it at the moment. So let
s just leave it as is. Though actually maybe we can get rid of the `getattr` and just do `device.name == "default.mixed"`.